css/demo.css

```diff
@@ -0,0 +1,291 @@
+body {
+  display: grid;
+  grid-template: 'site-header' auto 'main' 1fr 'footer' auto / 100%;
+  margin: 0 var(--double-gap);
+  padding: var(--double-gap);
+}
+
+.site-header {
+  display: grid;
+  font-family: var(--sans-serif);
+  font-size: var(--small);
+  grid-area: site-header;
+  grid-template-columns: minmax(8ch, 15ch) auto;
+}
+
+#banner-logo {
+  &:link,
+  &:visited {
+    align-self: start;
+    display: block;
+    transition: transform 300ms cubic-bezier(0.175, 0.885, 0.32, 1.275);
+    transform-origin: top center;
+    z-index: 1;
+  }
+
+  &:hover,
+  &:focus,
+  &:active {
+    transform: scale(1.05);
+  }
+
+  [data-logo] {
+    block-size: auto;
+    inline-size: 100%;
+  }
+}
+
+[data-navlist] {
+  align-items: center;
+  display: flex;
+  flex-wrap: wrap;
+  font-size: var(--normal);
+  justify-content: var(--nav-placement, flex-end);
+  list-style: none;
+  margin-block: 0 var(--gap);
+  padding: 0;
+
+  @media (width >=40em) {
+    --nav-placement: start;
+  }
+}
+
+[data-nav] {
+  display: inline-block;
+  padding: 0 var(--gap);
+
+  &:not(:hover, :active, :focus, [aria-current]) {
+    --underline-color: transparent;
+    --underline-thickness: 0;
+  }
+}
+
+[data-layout='main'] {
+  grid-area: main;
+}
+
+.main-header {
+  container: main-header / inline-size;
+  position: relative;
+  justify-content: center;
+
+  @media (width >=70em) {
+    align-items: var(--header-items-align, center);
+    display: flex;
+    gap: var(--double-gap);
+  }
+
+  @media (width >=1400px) {
+    --header-items-align: last baseline;
+  }
+}
+
+#title {
+  display: grid;
+  grid-auto-rows: min-content;
+  line-height: 0.8;
+  margin-block: 0;
+
+  @media (width >=56em) {
+    align-content: center;
+  }
+
+  & :not(.subtitle) {
+    color: var(--feature-to);
+    text-transform: uppercase;
+
+    @supports (background-clip: text) {
+      & {
+        background: var(--brand-pink);
+        background: var(--feature-gradient);
+        background-clip: text;
+        -webkit-box-decoration-break: clone;
+        box-decoration-break: clone;
+        color: transparent;
+        text-shadow: none;
+      }
+    }
+  }
+}
+
+[data-popover-title] {
+  font-size: var(--title-size);
+
+  @container main-header (width >=350px) {
+    --title-size: var(--step-8);
+  }
+
+  @container main-header (width >=1200px) {
+    --title-size: var(--step-9);
+  }
+
+  @container main-header (width >=1650px) {
+    --title-size: var(--step-10);
+  }
+}
+
+[data-popover-title~='attribute'] {
+  font-weight: 400;
+}
+
+.subtitle {
+  font-family: var(--serif);
+  font-style: italic;
+  font-weight: 400;
+}
+
+.description {
+  max-inline-size: 65ch;
+
+  @container main-header (width >=830px) {
+    min-inline-size: 35ch;
+  }
+}
+
+.demo-section {
+  /* container: demo-section / inline-size; */
```

Comment:
@dvdherron This seems to be the specific CSS that's messing with the placement of the shadow popover, when the shadow root is a descendant of `.demo-section`. Any idea what's causing that, or whether we can remove this line or something?

Reply:
Strange. I looked at this in an older version of Firefox too and was able to see the weird placement with containment turned on. It seems like for some reason applying containment to this section changes the positioning context, but only for the popovers in that are fully in the shadow dom. 

I was only using containment here for sizing, so it's not a big deal to remove containment from the `.demo-section`. I removed it in https://github.com/oddbird/popover-polyfill/pull/187/commits/3252bb661d9070dc25f249eef02e38735610551e